LeaderboardBackend.Test/Features/Users/ResetPasswordTests.cs

```diff
@@ -0,0 +1,321 @@
+        BCryptNet.EnhancedVerify("P4ssword", recovery.User.Password).Should().BeTrue();
+    }
+
+    [Test]
+    public async Task ResetPassword_Banned()
+    {
+        ApplicationContext context = _scope.ServiceProvider.GetRequiredService<ApplicationContext>();
+        int userNumber = _userNumber++;
+
+        AccountRecovery recovery = new()
+        {
+            CreatedAt = Instant.FromUnixTimeSeconds(20),
+            ExpiresAt = Instant.FromUnixTimeSeconds(20) + Duration.FromHours(1),
+            User = new()
+            {
+                Email = $"pwdresettestuser{userNumber}@email.com",
+                Password = BCryptNet.EnhancedHashPassword("P4ssword"),
+                Username = $"pwdresettestuser{userNumber}",
+                Role = UserRole.Banned
+            }
+        };
+
+        context.AccountRecoveries.Add(recovery);
+        await context.SaveChangesAsync();
+
+        HttpResponseMessage res = await _client.PostAsJsonAsync(Routes.RecoverAccount(recovery.Id), new ChangePasswordRequest
+        {
+            Password = "AValidP4ssword"
+        });
+
+        res.Should().HaveStatusCode(System.Net.HttpStatusCode.Forbidden);
+        context.ChangeTracker.Clear();
+        recovery = await context.AccountRecoveries.Include(ar => ar.User).SingleAsync(ar => ar.Id == recovery.Id);
+        recovery.UsedAt.Should().BeNull();
+        BCryptNet.EnhancedVerify("P4ssword", recovery.User.Password).Should().BeTrue();
+    }
+
+    [TestCase("OuxNzURtWdXWd", Description = "No number")]
+    [TestCase("DZWVZVV5ED8QE", Description = "No lowercase letter")]
+    [TestCase("y267pmi50skcc", Description = "No uppercase letter")]
+    [TestCase("zmgoyGS", Description = "7 characters")]
+    [TestCase("qutOboNSzYplEKCDlCEbGPIEtMEnJImHwnluHvksTZbhuHSwFLpvUZQQxIdHctldJkdEVMRyiWcyuIeBe",
+        Description = "81 characters")]
+    public async Task ResetPassword_BadPassword(string pwd)
+    {
+        ApplicationContext context = _scope.ServiceProvider.GetRequiredService<ApplicationContext>();
+        int userNumber = _userNumber++;
+
+        AccountRecovery recovery = new()
+        {
+            CreatedAt = Instant.FromUnixTimeSeconds(20),
+            ExpiresAt = Instant.FromUnixTimeSeconds(20) + Duration.FromHours(1),
+            User = new()
+            {
+                Email = $"pwdresettestuser{userNumber}@email.com",
+                Password = BCryptNet.EnhancedHashPassword("P4ssword"),
+                Username = $"pwdresettestuser{userNumber}",
+                Role = UserRole.Confirmed
+            }
+        };
+
+        context.AccountRecoveries.Add(recovery);
+        await context.SaveChangesAsync();
+
+        HttpResponseMessage res = await _client.PostAsJsonAsync(Routes.RecoverAccount(recovery.Id), new ChangePasswordRequest
+        {
+            Password = pwd
+        });
+
+        res.Should().HaveStatusCode(System.Net.HttpStatusCode.UnprocessableEntity);
```

Comment:
Must also assert that the `PasswordFormat` error code is returned.

You can pretty much copy/paste from this existing registration test: https://github.com/leaderboardsgg/leaderboard-backend/blob/main/LeaderboardBackend.Test/Features/Users/RegistrationTests.cs#L94-L99